### DISCUSSION TITLE
Mount Commands

### DISCUSSION THREAD

**Discussion by CBB3810** [pglombardo/PasswordPusher#1363] [Q&A] [answered]
On july 17 I was working with an ephemeral:release image that let me mount external directories and I was able to personalize the UI. Yesterday I updated the image with a pull command and now I can't mount the very same directories.
I was using this run command:

docker run -d -e PWP__THEME=darkly -e PWP_PRECOMPILE=true -p 0.0.0.0:5100:5100 -v /opt/dockerdata/logos:/opt/PasswordPusher/public/logos --mount type=bind,source=/opt/dockerdata/yaml/settings.yml,target=/opt/PasswordPusher/config/settings.yml pglombardo/pwpush-ephemeral:release

If I pull the mount commands and run the container without the -v and --mount, the image fires up in it's default state without any issues. 
I did verify that my ownership of the directory and files within the external directories are owned by the same user starting the container.

**Comment by pglombardo** [comment]
Hi @CBB3810 - Nothing has changed in terms of directory structure.

What's the error/issue when you have the mount commands?

**Reply by CBB3810** [reply]
the container just won't start at all?

**Reply by CBB3810** [reply]
I even went to the extreme and completely uninstalled my docker services from the server and deleted /etc/docker.. I wiped out everything and reinstalled. The only thing I have not tried is doing it on a fresh platform.

**Reply by pglombardo** [reply]
Nah don't use the nuclear option. :-)

> the container just won't start at all?

What's the output?

**Reply by CBB3810** [reply]
Are you asking in terms of docker logs <container_id> ?

**Reply by pglombardo** [reply]
That would help too but overall is there a docker error?

**Reply by pglombardo** [reply]
Remember that you have `PWP_PRECOMPILE` set so it may take 30-90 seconds to boot while it re-compiles all of the assets.

**Reply by CBB3810** [reply]
I don't see any errors hit my console when I execute my 'docker run' command, and to reiterate, it's something to do with the mounts because when I run the docker container with a vanilla command that doesn't involve external mounts, it works fine like this:
docker run -d -e PWP__THEME=darkly -e PWP_PRECOMPILE=true -p 0.0.0.0:5100:5100 pglombardo/pwpush-ephemeral:release

**Reply by CBB3810** [reply]
I'm aware about the precompile option but that's not it, because it literally bails out on starting the container if I put in the mount commands.

**Reply by pglombardo** [reply]
What does "bails out" mean?  That error should give us a clue.

**Reply by CBB3810** [reply]
I'm sorry, slang words. 
the docker service doesn't successfully start the container. I ran: journalctl -xeu docker  I suspect something is misconfigured in my firewallD

Aug 17 09:04:49 portal2 dockerd[11010]: time="2023-08-17T09:04:49.746827474-04:00" level=info msg="ignoring event" container=011dcc2ab84f66772bbbf252f7f750240ebb946e43b29ac1f246190ecf085805 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Aug 17 09:06:45 portal2 dockerd[11010]: time="2023-08-17T09:06:45.997849555-04:00" level=info msg="Firewalld: interface docker0 already part of docker zone, returning"
Aug 17 09:06:46 portal2 dockerd[11010]: time="2023-08-17T09:06:46.011963487-04:00" level=info msg="Firewalld: interface docker0 already part of docker zone, returning"
Aug 17 09:07:04 portal2 dockerd[11010]: time="2023-08-17T09:07:04.164990890-04:00" level=info msg="ignoring event" container=3849bd88e1b0cbb856661d2f810f8ece60e53f81f078c05142ead515b9ea3e25 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Aug 17 09:07:33 portal2 dockerd[11010]: time="2023-08-17T09:07:33.913484582-04:00" level=info msg="Firewalld: interface docker0 already part of docker zone, returning"
Aug 17 09:07:33 portal2 dockerd[11010]: time="2023-08-17T09:07:33.928886369-04:00" level=info msg="Firewalld: interface docker0 already part of docker zone, returning"
Aug 17 09:14:24 portal2 dockerd[11010]: time="2023-08-17T09:14:24.922836243-04:00" level=info msg="Container failed to exit within 10s of signal 15 - using the force" container=947b3c0924af2314fb31f0a347976bdfbdd81d4b4820a1629b4a52759d6bed06
Aug 17 09:14:24 portal2 dockerd[11010]: time="2023-08-17T09:14:24.972689355-04:00" level=info msg="ignoring event" container=947b3c0924af2314fb31f0a347976bdfbdd81d4b4820a1629b4a52759d6bed06 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Aug 17 09:18:13 portal2 dockerd[11010]: time="2023-08-17T09:18:13.437221568-04:00" level=info msg="ignoring event" container=06edb49c034633850693bc6cb4274bec355bb1c31c7593bd15f739fbe85b977a module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Aug 17 09:34:14 portal2 dockerd[11010]: time="2023-08-17T09:34:14.815055051-04:00" level=info msg="ignoring event" container=2404479ccae44fdd1b69106aa26ec752e242cb5143adb2c3e9cb6b398b7e5279 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Aug 17 09:35:35 portal2 dockerd[11010]: time="2023-08-17T09:35:35.930533681-04:00" level=info msg="ignoring event" container=63025d419cf7dd29ae8c6829806e9c3ab7b71aa02dd6e68cc3eceda1ca2d8e21 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Aug 17 09:40:57 portal2 dockerd[11010]: time="2023-08-17T09:40:57.775898396-04:00" level=info msg="ignoring event" container=87f8de50f25912695bff3898fb8252322a1fb868fc970c106ca035e4516177d8 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"

**Reply by CBB3810** [reply]
That doesn't seem to make any sense though. Because I can indeed start the container without any mount commands associated and connect to it.

**Reply by pglombardo** [reply]
> the docker service doesn't successfully start the container.

Whats the error?  Can you post the docker container output?

**Reply by CBB3810** [reply]
`[dockerdaemon@portal2 ~]$ docker inspect youthful_bouman
[
    {
        "Id": "2fcd5f3a0ad8d6832773f457a3c9072319ed3dc2c0a22cabab47e596a0e5f72c",
        "Created": "2023-08-17T14:49:19.836184924Z",
        "Path": "containers/docker/pwpush-ephemeral/entrypoint.sh",
        "Args": [],
        "State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "",
            "StartedAt": "2023-08-17T14:49:20.053240958Z",
            "FinishedAt": "2023-08-17T14:49:22.48995121Z"
        },
        "Image": "sha256:597fa92df491ca6052108709003652eb5649d1fb2af2d6ecf8e94448500bf5cc",
        "ResolvConfPath": "/opt/docker/containers/2fcd5f3a0ad8d6832773f457a3c9072319ed3dc2c0a22cabab47e596a0e5f72c/resolv.conf",
        "HostnamePath": "/opt/docker/containers/2fcd5f3a0ad8d6832773f457a3c9072319ed3dc2c0a22cabab47e596a0e5f72c/hostname",
        "HostsPath": "/opt/docker/containers/2fcd5f3a0ad8d6832773f457a3c9072319ed3dc2c0a22cabab47e596a0e5f72c/hosts",
        "LogPath": "/opt/docker/containers/2fcd5f3a0ad8d6832773f457a3c9072319ed3dc2c0a22cabab47e596a0e5f72c/2fcd5f3a0ad8d6832773f457a3c9072319ed3dc2c0a22cabab47e596a0e5f72c-json.log",
        "Name": "/youthful_bouman",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/opt/dockerdata/logos:/opt/PasswordPusher/public/logos"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {
                "5100/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "5100"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "ConsoleSize": [
                45,
                125
            ],
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "private",
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": [],
            "BlkioDeviceWriteBps": [],
            "BlkioDeviceReadIOps": [],
            "BlkioDeviceWriteIOps": [],
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": null,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "Mounts": [
                {
                    "Type": "bind",
                    "Source": "/opt/dockerdata/yaml/settings.yml",
                    "Target": "/opt/PasswordPusher/config/settings.yml"
                }
            ],
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/opt/docker/overlay2/425f2519301af8b0460c005c02c42e9e8ffb3d5479f9727697ff4985afe08744-init/diff:/opt/docker/overlay2/c8ea88f45f7fd34c12f5749a2cd2d522342a77d6e5f5b0970759ee40d38c52d7/diff:/opt/docker/overlay2/8be8cf923e67aa781ad78d5e8af22689a1b519a575a09f4afeee82222dfa9cfe/diff:/opt/docker/overlay2/4d3f3e63526eb4a2cb28562cf7400710abbcc5d2b6a1ec64ce7748f6c9e312c9/diff:/opt/docker/overlay2/aff87029f1a1b3637b834ee13dda90e57d26d40ad90b560543e7eee0c3dfea60/diff:/opt/docker/overlay2/e5ff746dca84b363f55c60b0ede08269e7e7a430b3b495a6d7d678662a241e79/diff:/opt/docker/overlay2/c67c1b8dc5b7cd09d3d04ee068d55c33c274a1e08869ae2d30f9d82acc310d3c/diff:/opt/docker/overlay2/5003c1a6ccbbc2e572fba708ff796c7d8b11599e1856b753f83c1ed1fafaf109/diff:/opt/docker/overlay2/d3178dd71576b5facb710686ea56d0c25e5724944feb6abe08818e540ac82596/diff:/opt/docker/overlay2/8b5f4485804d7f7bc871534f00c4b253c1b075bfde89e833556c7055cf1518e8/diff:/opt/docker/overlay2/4f983fac08e03f3ad56b14efd7402394e6f038b5b8f6f8f6aa38769aab06dcda/diff:/opt/docker/overlay2/361f32947af4d4f9d83eb33a85392bd7c95d7b4dac5d09508b39824ae4e73870/diff",
                "MergedDir": "/opt/docker/overlay2/425f2519301af8b0460c005c02c42e9e8ffb3d5479f9727697ff4985afe08744/merged",
                "UpperDir": "/opt/docker/overlay2/425f2519301af8b0460c005c02c42e9e8ffb3d5479f9727697ff4985afe08744/diff",
                "WorkDir": "/opt/docker/overlay2/425f2519301af8b0460c005c02c42e9e8ffb3d5479f9727697ff4985afe08744/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/opt/dockerdata/logos",
                "Destination": "/opt/PasswordPusher/public/logos",
                "Mode": "",
                "RW": true,
                "Propagation": "rslave"
            },
            {
                "Type": "bind",
                "Source": "/opt/dockerdata/yaml/settings.yml",
                "Destination": "/opt/PasswordPusher/config/settings.yml",
                "Mode": "",
                "RW": true,
                "Propagation": "rslave"
            }
        ],
        "Config": {
            "Hostname": "2fcd5f3a0ad8",
            "Domainname": "",
            "User": "pwpusher",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "5100/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PWP__THEME=darkly",
                "PWP_PRECOMPILE=true",
                "PATH=:/usr/local/bundle/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "LANG=C.UTF-8",
                "RUBY_MAJOR=3.2",
                "RUBY_VERSION=3.2.2",
                "RUBY_DOWNLOAD_SHA256=4b352d0f7ec384e332e3e44cdbfdcd5ff2d594af3c8296b5636c710975149e23",
                "GEM_HOME=/usr/local/bundle",
                "BUNDLE_SILENCE_ROOT_WARNING=1",
                "BUNDLE_APP_CONFIG=/usr/local/bundle",
                "LC_CTYPE=UTF-8",
                "LC_ALL=en_US.UTF-8",
                "APP_ROOT=/opt/PasswordPusher",
                "HOME=",
                "RACK_ENV=private",
                "RAILS_ENV=private",
                "DATABASE_URL=sqlite3:db/db.sqlite3"
            ],
            "Cmd": null,
            "Image": "pglombardo/pwpush-ephemeral:release",
            "Volumes": null,
            "WorkingDir": "/opt/PasswordPusher",
            "Entrypoint": [
                "containers/docker/pwpush-ephemeral/entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "maintainer": "pglombardo@hey.com",
                "org.opencontainers.image.created": "2023-08-03T12:49:51.817Z",
                "org.opencontainers.image.description": "🔐   An application to securely communicate passwords over the web.  Passwords automatically expire after a certain number of views and/or time has passed.  Track who, what and when.",
                "org.opencontainers.image.licenses": "Apache-2.0",
                "org.opencontainers.image.revision": "5847b8de8e6579f1a53d8d6c620e2b152028301a",
                "org.opencontainers.image.source": "https://github.com/pglombardo/PasswordPusher",
                "org.opencontainers.image.title": "PasswordPusher",
                "org.opencontainers.image.url": "https://github.com/pglombardo/PasswordPusher",
                "org.opencontainers.image.version": "release"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "e6d649a8f4d9de36cc7670a0e2d0974f31164d86dc0318dcc761106616b76624",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/e6d649a8f4d9",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "d67bf9d088ca6d15ae8d7564d79130134b956c1e63813f748251393f9ff282d9",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]
`

**Reply by CBB3810** [reply]
apologies if I didn't post that correctly. That's my output from docker inspect.

**Reply by CBB3810** [reply]
I'm not sure of what output you were asking for. If there's something different I'm certainly happy to provide. Please just let me know exactly the command you'd like run and I'll capture / share the output.

**Comment by CBB3810** [comment]
`sudo cat /opt/docker/containers/2fcd5f3a0ad8d6832773f457a3c9072319ed3dc2c0a22cabab47e596a0e5f72c/2fcd5f3a0ad8d6832773f457a3c9072319ed3dc2c0a22cabab47e596a0e5f72c-json.log
[sudo] password for cboley: 
{"log":"Password Pusher: migrating database to latest...\n","stream":"stdout","time":"2023-08-17T14:49:20.053481422Z"}
{"log":"rake aborted!\n","stream":"stderr","time":"2023-08-17T14:49:22.478238295Z"}
{"log":"TypeError: no implicit conversion of nil into String\n","stream":"stderr","time":"2023-08-17T14:49:22.478270205Z"}
{"log":"/opt/PasswordPusher/config/initializers/devise.rb:187:in `initialize'\n","stream":"stderr","time":"2023-08-17T14:49:22.478370977Z"}
{"log":"/opt/PasswordPusher/config/initializers/devise.rb:187:in `new'\n","stream":"stderr","time":"2023-08-17T14:49:22.478383047Z"}
{"log":"/opt/PasswordPusher/config/initializers/devise.rb:187:in `block in \u003cmain\u003e'\n","stream":"stderr","time":"2023-08-17T14:49:22.478386847Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/devise-4.9.2/lib/devise.rb:314:in `setup'\n","stream":"stderr","time":"2023-08-17T14:49:22.478405928Z"}
{"log":"/opt/PasswordPusher/config/initializers/devise.rb:11:in `\u003cmain\u003e'\n","stream":"stderr","time":"2023-08-17T14:49:22.478409268Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/engine.rb:667:in `load'\n","stream":"stderr","time":"2023-08-17T14:49:22.478412468Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/engine.rb:667:in `block in load_config_initializer'\n","stream":"stderr","time":"2023-08-17T14:49:22.478415498Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/activesupport-7.0.6/lib/active_support/notifications.rb:208:in `instrument'\n","stream":"stderr","time":"2023-08-17T14:49:22.478418718Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/engine.rb:666:in `load_config_initializer'\n","stream":"stderr","time":"2023-08-17T14:49:22.478421668Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/engine.rb:620:in `block (2 levels) in \u003cclass:Engine\u003e'\n","stream":"stderr","time":"2023-08-17T14:49:22.478424968Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/engine.rb:619:in `each'\n","stream":"stderr","time":"2023-08-17T14:49:22.478428778Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/engine.rb:619:in `block in \u003cclass:Engine\u003e'\n","stream":"stderr","time":"2023-08-17T14:49:22.478432248Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/initializable.rb:32:in `instance_exec'\n","stream":"stderr","time":"2023-08-17T14:49:22.478435648Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/initializable.rb:32:in `run'\n","stream":"stderr","time":"2023-08-17T14:49:22.478438728Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/initializable.rb:61:in `block in run_initializers'\n","stream":"stderr","time":"2023-08-17T14:49:22.478442488Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/initializable.rb:50:in `each'\n","stream":"stderr","time":"2023-08-17T14:49:22.478446028Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/initializable.rb:50:in `tsort_each_child'\n","stream":"stderr","time":"2023-08-17T14:49:22.478460099Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/initializable.rb:60:in `run_initializers'\n","stream":"stderr","time":"2023-08-17T14:49:22.478463839Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/application.rb:372:in `initialize!'\n","stream":"stderr","time":"2023-08-17T14:49:22.478466969Z"}
{"log":"/opt/PasswordPusher/config/environment.rb:5:in `\u003cmain\u003e'\n","stream":"stderr","time":"2023-08-17T14:49:22.478470109Z"}
{"log":"\u003cinternal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb\u003e:37:in `require'\n","stream":"stderr","time":"2023-08-17T14:49:22.478473979Z"}
{"log":"\u003cinternal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb\u003e:37:in `require'\n","stream":"stderr","time":"2023-08-17T14:49:22.478477109Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'\n","stream":"stderr","time":"2023-08-17T14:49:22.478491159Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/zeitwerk-2.6.11/lib/zeitwerk/kernel.rb:38:in `require'\n","stream":"stderr","time":"2023-08-17T14:49:22.478494799Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/application.rb:348:in `require_environment!'\n","stream":"stderr","time":"2023-08-17T14:49:22.47849778Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/railties-7.0.6/lib/rails/application.rb:506:in `block in run_tasks_blocks'\n","stream":"stderr","time":"2023-08-17T14:49:22.47850084Z"}
{"log":"/opt/PasswordPusher/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/exe/rake:27:in `\u003ctop (required)\u003e'\n","stream":"stderr","time":"2023-08-17T14:49:22.47850386Z"}
{"log":"/usr/local/bin/bundle:25:in `load'\n","stream":"stderr","time":"2023-08-17T14:49:22.47850683Z"}
{"log":"/usr/local/bin/bundle:25:in `\u003cmain\u003e'\n","stream":"stderr","time":"2023-08-17T14:49:22.47850969Z"}
{"log":"Tasks: TOP =\u003e db:migrate =\u003e db:load_config =\u003e environment\n","stream":"stderr","time":"2023-08-17T14:49:22.47851261Z"}
{"log":"(See full trace by running task with --trace)\n","stream":"stderr","time":"2023-08-17T14:49:22.47851583Z"}
`

**Reply by pglombardo** [reply]
Ok this helps.  It looks like a problem with the `settings.yml`.

Try this command:

`docker run -d -e PWP__THEME=darkly -e PWP_PRECOMPILE=true -p 0.0.0.0:5100:5100 -v /opt/dockerdata/logos:/opt/PasswordPusher/public/logos --mount type=bind,source=/opt/dockerdata/yaml/settings.yml,target=/opt/PasswordPusher/config/settings.yml pglombardo/pwpush-ephemeral:1.30.12`

**Reply by CBB3810** [reply]
0d05e9705515   pglombardo/pwpush-ephemeral:1.30.12   "containers/docker/p…"   13 seconds ago   **Exited (1) 8 seconds ago**              magical_liskov

**Reply by CBB3810** [reply]
Do you want me to post my settings.yaml?

**Reply by pglombardo** [reply]
Hi @CBB3810 - did you ever find a solution to this issue?  From that stack trace alone, it looks like there is an issue with the settings.yaml [specifically with this line](https://github.com/pglombardo/PasswordPusher/blob/master/config/settings.yml#L91).  Could you try adding that line and seeing what affect it has?

**Comment by CBB3810** [comment]
[settings.yml.txt](https://github.com/pglombardo/PasswordPusher/files/12371168/settings.yml.txt)

**Reply by pglombardo** [reply]
Well a preliminary update.  I booted a container with your `settings.yml` and besides a broken logo because of the missing directory mounts, there was no issue.

Really odd/not sure.  Give me some time to think about it.

**Reply by CBB3810** [reply]
Let me ask a question for clarity. If I am running as user "dockerdaemon" and I've made the external mount directories owned by dockerdaemon:dockerdaemon with 0755 permissions I should be fine as far as mounts go. Correct?

**Reply by pglombardo** [reply]
I would think so but you could try to `chmod 777` just to rule that possibility out.

In the docker inspect you posted, you are getting a `TypeError: no implicit conversion of nil into String` which is causing the docker container to fail to start.  It's almost as if the `release` image that you have isn't complete.  But testing it here locally and it works.  And no other reports of this...

Did you try another tag?

**Reply by CBB3810** [reply]
please suggest a tag and I'll grab it.

**Reply by pglombardo** [reply]
Try `1.30.12` and `latest`

**Reply by CBB3810** [reply]
I think this troubleshooting will go faster if I just fire up a new VM and try this out with a fresh OS. I'll let you know what I find.

**Reply by CBB3810** [reply]
I tried the new tags. Same behavior.

**Reply by CBB3810** [reply]
I want to provide some specifics:
I'm running rocky linux 9 with docker installed from the official https://download.docker.com/linux/centos/docker-ce.repo
I put a dockerdata directory within the root of the filesystem "/dockerdata"
I then changed ownership on that directory dockerdaemon:dockerdaemon
after I did that I ran as dockerdaemon: 
docker run -d -v /dockerdata/pwpush/logos:/opt alpine:latest tail -f /dev/null
docker container ls -a
CONTAINER ID   IMAGE                COMMAND                  CREATED             STATUS                         PORTS     NAMES
123d87dcb394   alpine:latest      "tail -f /dev/null"      14 seconds ago      Up 13 seconds                            condescending_panini
docker exec -it condescending_panini /bin/sh 
/ # ls /opt
JOYND_Logo_2022.png  gcr-gnupg.png        gcr-password.png     pinpng.png

These are the correct files in the external directory.

The alpine container didn't stop. It mounted the external volume, and I can list the files within the external directory.

I'm led to believe that the "clean new system" I'm using is working, and all docker components are working as they should.

I would suggest that it might be the settings.yml but I took a stock settings.yml downloaded from your site replaced that one in my setup which means the directory would/should be mounted to the container but the current ephemeral:release version won't work for me.

Based on a process of elimination, I'm left with no other logical explanation than to believe something on the current ephemeral:release version won't let the volume nor the file mount and this is the reason why the ephemeral:release container completely fails to start.

All I can remember is that on July 17th is when I got my previous setup working. I believe I had a few github posts at that time too.
At that time, I was also using the "ephemeral:release" version.
How can I find that particular release number from July 17th so I can try and test it please?

**Reply by CBB3810** [reply]
Greetings Peter, this works on the original system. 1.29.1 does not behave in the same as the current ephemeral:
docker run -d -e PWP__THEME=darkly -e PWP_PRECOMPILE=true -p 0.0.0.0:5100:5100 -v /opt/dockerdata/logos:/opt/PasswordPusher/public/logos --mount type=bind,source=/opt/dockerdata/yaml/settings.yml,target=/opt/PasswordPusher/config/settings.yml pglombardo/pwpush-ephemeral:**1.29.1**

**Comment by CBB3810** [comment, marked as answer]
I have found that the last working version is 1.29.3. Afterwards I can't get the container to start when mounting external files. Whatever changed happened between that version and 1.30.0

**Reply by CBB3810** [reply]
Please let me know if you can confirm the same Peter or it's just docker on Rocky Linux 9 that this is happening.

**Reply by pglombardo** [reply]
Hi @CBB3810 - apologies I've been tied up with work.  II'll try myself again with mounts over the weekend.

**Reply by CBB3810** [reply]
No worries. Mine is working perfectly again and I appreciate the use of your work.

**Reply by CBB3810** [reply]
Hi Peter, did you ever find that this was a reproducible error?

**Reply by pglombardo** [reply]
Hi @CBB3810 - apologies that I've been overloaded.  I am going to try and find some time to retry this today.
